### ISSUE TITLE
implement Linux docker launcher

### ISSUE THREAD

**Issue by stebourbi** [stebourbi/sbt-docker-launcher#1]
On linux docker needs root user abilitation to run commands and this is an issue in a context of an sbt build. I have not found any statisfying solution to resolve that but only some workarounds. And I think that we should implement all of the available workarounds and let the user decide which one to implement.
Here are the workarounds that i have found:
-  prompt for sudo password each time a command is launched 
- add the current user to the docker group
- pass the sudo password to the commands via a configuration (system conf or a file)
- generate a  script that runs the commands and define it as passwordless in sudo configuration

there are probably bettter solutions but that's what i have found actually!

**Comment by sjeandeaux** [comment]
Question and potentials actions by workarounds : 

I order workaround by prefered implementation.

##add the current user to the docker group
- Detect if user is in group **docker**
- Modify scala code add LinuxDocker without sudo
- Create page wiki to indicate how add user in groud docker

##pass the sudo password to the commands via a configuration (system conf or a file)
- Modify scala to read this file
- What is localisation of file (~/.sbt-docker-launcher or ...)?

##prompt for sudo password each time a command is launched
- Modify scala code to integrate a prompt

##generate a script that runs the commands and define it as passwordless in sudo configuration
- when do you run this commands?

**Comment by stebourbi** [comment]
I agree with that. Let's start implementation. And will see the result.

**Comment by stebourbi** [comment]
I mean start with the first one :grimacing:

**Comment by sjeandeaux** [comment]
i keep "prompt solution" before i create a Dockerfile with docker on.
